src/main/java/com/spotify/autoscaler/ClusterStats.java

```diff
@@ -211,62 +107,43 @@ void setStats(BigtableCluster cluster, int currentNodes) {
 
     if (clusterData == null) {
       // First time we saw this cluster, register a gauge
-      this.registry.register(
-          APP_PREFIX
-              .tagged("what", "node-count")
-              .tagged("project-id", cluster.projectId())
-              .tagged("cluster-id", cluster.clusterId())
-              .tagged("instance-id", cluster.instanceId()),
-          (Gauge<Integer>)
-              () -> registeredClusters.get(cluster.clusterName()).getCurrentNodeCount());
 
-      this.registry.register(
-          APP_PREFIX
-              .tagged("what", "min-node-count")
-              .tagged("project-id", cluster.projectId())
-              .tagged("cluster-id", cluster.clusterId())
-              .tagged("instance-id", cluster.instanceId()),
-          (Gauge<Integer>) () -> registeredClusters.get(cluster.clusterName()).getMinNodeCount());
-
-      this.registry.register(
-          APP_PREFIX
-              .tagged("what", "max-node-count")
-              .tagged("project-id", cluster.projectId())
-              .tagged("cluster-id", cluster.clusterId())
-              .tagged("instance-id", cluster.instanceId()),
-          (Gauge<Integer>) () -> registeredClusters.get(cluster.clusterName()).getMaxNodeCount());
-
-      this.registry.register(
-          APP_PREFIX
-              .tagged("what", "effective-min-node-count")
-              .tagged("project-id", cluster.projectId())
-              .tagged("cluster-id", cluster.clusterId())
-              .tagged("instance-id", cluster.instanceId()),
-          (Gauge<Integer>)
-              () -> registeredClusters.get(cluster.clusterName()).getEffectiveMinNodeCount());
-
-      this.registry.register(
-          APP_PREFIX
-              .tagged("what", "last-check-time")
-              .tagged("project-id", cluster.projectId())
-              .tagged("cluster-id", cluster.clusterId())
-              .tagged("instance-id", cluster.instanceId()),
-          (Gauge<Long>)
-              () ->
-                  db.getBigtableCluster(
-                          cluster.projectId(), cluster.instanceId(), cluster.clusterId())
-                      .flatMap(
-                          p ->
-                              Optional.of(
-                                  Duration.between(
-                                      p.lastCheck().orElse(Instant.EPOCH), Instant.now())))
-                      .get()
-                      .getSeconds());
+      for (BigtableMetric.Metrics metric : BigtableMetric.Metrics.values()) {
+        // This will return null for non implemented getMetricValues. This will happen when the
+        // metric depends in other things other than the registeredClusters or the cluster
+        // itself, e.g., when it is dependent on the database.
+        Gauge metricValue = metric.getMetricValue(registeredClusters.get(cluster.clusterName()));
+        switch (metric) {
+          case LAST_CHECK_TIME:
```

Comment:
i don't like this here :)
i can't think of anything other than passing the `db` to the enum, but even that would be better than this i think.
any other opinions?

Reply:
I don't like it neither, I'm trying to think in a better solution...
My initial approach leave the metrics implementation in the ClusterStats was to avoid sending too many contextual objects to the Metrics class... :( 

Still thinking on how to improve this... :)

Reply:
I added the db, I'm not feeling so bad as I thought I would feel with this change... :)